Fix applying variable updates provided by plugins (#9437)
eclipse-theia/theia: CHANGELOG.md

# Change Log

## v1.14.0

<a name="breaking_changes_1.14.0">[Breaking Changes:](#breaking_changes_1.14.0)</a>

> - [terminal] `ShellProcess` does not longer merge the received `ShellProcessOptions` with the process environment variables.  

Reviewer: ```suggestion
- [terminal] `ShellProcess` no longer merges the received `ShellProcessOptions` with the process environment variables.  
```

Author: After Paul's comments, I removed the breaking changes, see: 
https://github.com/eclipse-theia/theia/pull/9437#discussion_r627587900
